### PR TITLE
Fix/2403

### PR DESCRIPTION
**Pull request by diya2214**

### CODE REVIEW

File: app/helpers/handlers.js [hunk 1]
```diff
@@ -383,22 +373,24 @@ const showPostPage = (currentQuestion, request, h) => {
...
+      } else if (getYarValue(request, 'calculatedGrantCalf') >= 500000) {
         setYarValue(request, 'calculatedGrant', 500000)
         return h.redirect('/upgrading-calf-housing/potential-amount-conditional')
         // solar capping
       } else if (getYarValue(request, 'calculatedGrant') > 500000) {
-        const newCap = 500000 - getYarValue(request, 'calculatedGrantCalf')
+        const solarCap = 500000 - getYarValue(request, 'calculatedGrantCalf')
         // store capped solar value for potential amount solar capped page
-        setYarValue(request, 'calculatedGrantSolarPreCap', getYarValue(request, 'calculatedGrantSolar'))
+        setYarValue(request, 'calculatedGrantSolarPreCap', getYarValue(request, 'SolarPVCost') * 0.25)
```
Comment (zavies97) [inline app/helpers/handlers.js:387]: why are you calculating this value when we have the original calculatedSolarGrant value?

File: app/helpers/handlers.js [hunk 2]
```diff
@@ -383,22 +373,24 @@ const showPostPage = (currentQuestion, request, h) => {
...
         // set cap for solar and grant
-        setYarValue(request, 'calculatedGrantSolar', newCap)
-        setYarValue(request, 'calculatedGrant', 500000)
+        if (getYarValue(request, 'calculatedGrantSolar') > solarCap) {
```
Comment (zavies97) [inline app/helpers/handlers.js:389]: we dont necessarily need the if statement, as the user would only end up in this section if solar has been capped

File: app/helpers/handlers.js [hunk 3]
```diff
@@ -383,22 +373,24 @@ const showPostPage = (currentQuestion, request, h) => {
...
+          setYarValue(request, 'calculatedGrantSolar', solarCap)
+          setYarValue(request, 'calculatedGrant', 500000)
+        }
+        setYarValue(request, 'solarCap', solarCap)
```
Comment (zavies97) [inline app/helpers/handlers.js:393]: if we are setting calculatedGrantSolar to the solarCap, why do we need a seperate yarKey for solarCap?

File: app/helpers/handlers.js (hunk 2 above)
Comment (diya2214) [reply app/helpers/handlers.js:389]: yes it was not working when the cap is more than solar grant calculated

File: app/helpers/handlers.js (hunk 3 above)
Comment (diya2214) [reply app/helpers/handlers.js:393]: its in cases when cap is more than grant for solar

File: app/helpers/handlers.js [hunk 4]
```diff
@@ -50,24 +50,6 @@ const getPage = async (question, request, h) => {
   }
 
   switch (url) {
-  // reset environmentalImpact yar key if switching between pages
-    case 'roof-solar-PV':
```
Comment (zavies97) [inline app/helpers/handlers.js:54]: why was this removed?

Comment (diya2214) [reply app/helpers/handlers.js:54]: why it was there in the first place handled in payload depending on user response

File: app/helpers/handlers.js (hunk 1 above)
Comment (zavies97) [reply app/helpers/handlers.js:387]: it is true, yet calculatedGrantSolar is the original calculated grant with no capping, aka what the user would get. we set that to calculatedGrantSolarPreCap to show the value before the capping (the original value), then set calcualtedGrantSolar to be the users solar grant post capping, so we have both values

I'm just nto sure why you are redoing the same calculation when we already have this value :P

File: app/helpers/handlers.js (hunk 4 above)
Comment (zavies97) [reply app/helpers/handlers.js:54]: we added it as a user could potentially go through upgrading-existing-building etc, then go back to roof-solar-pv and go directly to solar-pv-system, so it was a way to reset all yarkeys on the branching paths

File: app/helpers/handlers.js (hunk 2 above)
Comment (zavies97) [reply app/helpers/handlers.js:389]: but.... if the cap is more than the solar grant calculated then this code wouldn't run at all...

File: app/helpers/handlers.js (hunk 1 above)
Comment (diya2214) [reply app/helpers/handlers.js:388]: we are not doing any calculation here just separating variables . if calculated grant will be capped based on rules . we still have to show uncapped value which is 25% of entered value

Comment (zavies97) [reply app/helpers/handlers.js:388]: we already do, its called calculatedgrantsolar

You're calculating it by mutliplying the users entered value by 0.25, thus doing the SAME CALCULATION a second time

Comment (diya2214) [reply app/helpers/handlers.js:388]: got your point changing code

File: app/helpers/handlers.js (hunk 2 above)
Comment (diya2214) [reply app/helpers/handlers.js]: removing if statement

File: app/config/question-bank.js [hunk 5]
```diff
@@ -1702,7 +1702,7 @@ const questionBank = {
             <p class="govuk-body">This grant amount combines:<p>
             <ul>
             <li>£{{_calculatedGrantCalf_}} for calf housing costs (40% of £{{_CalfHousingCost_}})</li>
-            <li>£{{_calculatedGrantSolar_}} for solar PV costs (25% of £{{_SolarPVCost_}} is £{{_calculatedGrantSolarPreCap_}}, capped at £{{_solarCap_}})</li>
+            <li>£{{_calculatedGrantSolar_}} for solar PV costs (25% of £{{_SolarPVCost_}} is £{{_calculatedGrantSolarPreCap_}}, capped at £{{_calculatedGrantSolar_}})</li>
             </ul>
             <p class="govuk-body">As calf housing costs take grant funding priority, you may be able to apply for a grant of up to £{{_solarCap_}} for solar PV system costs. The maximum grant is £500,000.`,
```
Comment (zavies97) [inline app/config/question-bank.js:1707]: solarCap here needs to be replaced with calculatedGrantSolar as well

File: app/helpers/handlers.js (hunk 3 above)
Comment (diya2214) [reply app/helpers/handlers.js]: changed

**Review by zavies97 — APPROVED** [review]

File: app/config/question-bank.js (hunk 5 above)
Comment (diya2214) [reply app/config/question-bank.js]: done
